### ISSUE TITLE
Adapting the code for sinus function

### ISSUE THREAD

**Issue by couchexpert** [makeyourownneuralnetwork/makeyourownneuralnetwork#62]
Hey Tariq, hey all,

I think I understand the code from the book quite well. I wonder now how to adapt this code for the prediction of e.g. the sinus function. I took the code from the book and choose following setup:

- input data is from 0 until 2*pi. Target is the sinus function
- 1 input neuron, 1 hidden layer with 10 neurons and 1 output neuron.
- training data is 2880 samples, shuffled.  test data is 500 samples (0 until 0.35*pi)
- input and target data are normalized between [0.01 and 0.99]
- e.g. the first original input data is 0.0001 and the target around 0.5
- for training purpose I pass one input and one target sample in the train.function and let the weights change. repeat this for all 2880 samples.
- learning rates between 0.000001 and 0.5
- epochs up to 10000

The challenge is, that the error function won´t decrease over time and the result of the sinus prediction is terrible. Is there any major weak point in my setup which I oversee? Can the original code easily be adapted to predict sinus?

**Comment by makeyourownneuralnetwork** [comment]
Hi @couchexpert 

Do you mean the sine / sin function? (I google sinus function and couldn't find it)

Usually the problem is of network architecture. Many of the questions I get are trying to achieve a complex function with a network that is too small to have sufficient learning capacity.

I did a quick google and people have succeeded in modelling sine/cosine .. eg 
[https://stats.stackexchange.com/questions/320289/approximate-the-sine-function-with-shallow-neural-network](https://stats.stackexchange.com/questions/320289/approximate-the-sine-function-with-shallow-neural-network)

Various other posts on the internet suggest that it is difficult to get a good approximation with a shallow network (even if mathematically possible). 

Perhaps try an additional layer? 

Sorry I couldn't explore this more right now. Let me know how your experiments go.
